ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -615,8 +614,8 @@ validateCandidate lgrDB tracer cfg invalidPoints
             candidate' = fromMaybe
               (error "cannot rollback to point on fragment") $
               AF.rollback lastValid candidate
-        let invalidPointsInCand = Set.fromList $ pointsStartingFrom pt
-        atomically $ modifyTVar' invalidPoints (Set.union invalidPointsInCand)
+        let invalidBlocksInCand = hashesStartingFrom pt
+        atomically $ modifyTVar' invalid (Map.union invalidBlocksInCand)
```

Comment:
Need to import `Data.Map.Strict` for this module, and probably all other modules in the package.